Assets/Scripts/UI/PauseMenu.cs

```diff
@@ -0,0 +1,47 @@
+﻿using TMPro;
+using UnityEngine;
+using UnityEngine.SceneManagement;
+
+namespace UI
+{
+    public class PauseMenu : Menu
+    {
+        [SerializeField] private GameObject pausePanel;
+
+        // Update is called once per frame
+        private void Update()
+        {
+            PauseGame();
+        }
+
+        public void SetHighScoreText()
+        {
+            var highScores = ScoreManager.GetScores();
+            var scoreTextBoxes = highScoreTextPanel.GetComponentsInChildren<TextMeshProUGUI>();
+            
+            SetHighScoreText(highScores, scoreTextBoxes);
+        }
+        
+        private void PauseGame()
+        {
+            if (Input.GetKeyDown(KeyCode.Escape))
+            {
+                Time.timeScale = 0;
+                pausePanel.SetActive(true);
+            }
+        }
+
+        public void UnpauseGame()
+        {
+            Time.timeScale = 1;
+            pausePanel.SetActive(false);
+        }
+
+        public void QuitGame()
+        {
+            Time.timeScale = 1;
+            pausePanel.SetActive(false);
+            SceneManager.LoadScene(0);
+        }
```

Comment:
Since this function return the the main menu, I suggest that the button in the pause menu is labeled "Quit to main menu" or something different than "Quit", because I think it is misleading, and one could expect the button to quit the application

Reply:
I renamed the text on the UI to `leave game` and renamed the function `Quit2MainMenu`